src/planscape/planning/permission.py

```diff
@@ -0,0 +1,40 @@
+from django.shortcuts import get_object_or_404
+from rest_framework.permissions import BasePermission
+from collaboration.permissions import (
+    PlanningAreaPermission,
+    ScenarioPermission,
+)
+from planning.models import PlanningArea
+
+
+class PlanscapePermission(BasePermission):
+    permission_set = None
+
+    def has_permission(self, request, view):
+        return self.is_authenticated(request)
+
+    def is_authenticated(self, request):
+        return request.user.is_authenticated
+
+    def has_object_permission(self, request, view, object):
+        if view.action == "update" or view.action == "partial_update":
+            return self.permission_set.can_change(request.user, object)
+        if view.action == "destroy":
+            return self.permission_set.can_remove(request.user, object)
+
+
+class PlanningAreaViewPermission(PlanscapePermission):
+    permission_set = PlanningAreaPermission
+
+
+class ScenarioViewPermission(PlanscapePermission):
+    permission_set = ScenarioPermission
+
+    def has_permission(self, request, view):
+        planningarea_pk = view.kwargs.get("planningarea_pk")
+        planningarea = get_object_or_404(PlanningArea, id=planningarea_pk)
+        if view.action == "create" and planningarea_pk:
+            return PlanningAreaPermission.can_add_scenario(request.user, planningarea)
+        if view.action == "list" and planningarea_pk:
+            return PlanningAreaPermission.can_view(request.user, planningarea)
+        return PlanningAreaPermission.can_change(request.user, planningarea)
```

Comment:
@george-silva  - let me know if I'm missing something here. 
For planning areas--update and delete permissions are checked for the specific object. We don't enforce anything at the PlanningAreaPermission level for creating new planning areas. For list action, DRF won't check here, so this is enforced at the query in the viewset.

For scenarios: we check for planning area permissions in the `has_permission` method for creating new scenarios and listing scenarios. And we fallback to the PlanscapePermission has_object_permission method for update and delete.